src/actions/assetsActions.js

```diff
@@ -709,3 +734,62 @@ export const resetLocalNonceToTransactionCountAction = (wallet: Object) => {
     dispatch(saveDbAction('txCount', { txCount: txCountNew }, true));
   };
 };
+
+export const speedUpTransactionAction = (transactionHash: string, gasPriceGwei: number, gasLimit: number) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const {
+      history: { data: history },
+      accounts: { data: accounts },
+      assets: { data: assets },
+    } = getState();
+    const accountAddress = getActiveAccountAddress(accounts);
+    const transactionInfo = (history[accountAddress] || []).find(
+      ({ hash }) => isCaseInsensitiveMatch(hash, transactionHash),
+    );
+    if (!transactionInfo) {
+      Toast.show({
+        message: 'Cannot speed up transaction',
+        type: 'warning',
+        title: 'Failed',
+        autoClose: false,
+      });
+      return;
+    }
+    const txFeeInWei = utils
+      .parseUnits(gasPriceGwei.toString(), 'gwei')
+      .mul(gasLimit);
+    const gasPrice = txFeeInWei.div(gasLimit).toNumber();
+    const {
+      to,
+      from,
+      asset: assetSymbol,
+      note,
+      value,
+    } = transactionInfo;
+    // $FlowFixMe
+    const asset: Asset = Object.values(assets).find(({ symbol }) => symbol === assetSymbol) || {};
```

Comment:
Can't we use `const asset = assets[assetSymbol];`?

In other hand, maybe we could handle the scenario where the asset does not exist? On that case, sending the user to the next screen sounds weird.

```js
const asset = assets[assetSymbol]; // (or whatever we use here)
if (!asset) {
  // ...
  return;
}
// ...

Reply:
Right, we should use `const asset = assets[assetSymbol];`